server/api/event/index.post.ts

```diff
@@ -0,0 +1,92 @@
+// POST /api/event
+// Endpoint for adding a new event to db
+
+export default defineEventHandler(async (event) => {
+  const { user } = event.context
+
+  const {
+    companyUID,
+    capacity,
+    date,
+    description,
+    limitedCapacity,
+    location,
+    title,
+    eventType,
+  } = await readBody(event)
+
+  // Check if user is authorized
+  if (hasAccess(user, ['company'])) {
+    if (user.companyUID !== companyUID)
+      throw createError({
+        statusCode: 401,
+        statusMessage: 'Company users cannot post events for other companies',
+      })
+  } else if (!hasAccess(user, ['admin']))
+    throw createError({
+      statusCode: 401,
+      statusMessage: 'User not authenticated',
+    })
+
+  // check if data is defined.
+  if (

```

Comment:
It would be nice if you only need to include the values that you want to update. The database ref function .update (that you use to update the data) will only update the key values that are defined. It will delete those that are set to null however, so probably smart to check for that. And also check that the values that are defined are valid of course.